R/post_ts.R

```diff
@@ -109,99 +112,20 @@ post_ts = function(files, on_exists=c("stop", "skip", "replace", "merge"), verbo
     # tag item with our special identifiers. if the item already existed,
     # identifiers should be wiped out by a known SB quirk, so sleep to give time
     # for the files to be added and the identifiers to disappear so we can replace them
-    for(wait in 1:100) {
-      Sys.sleep(0.2)
-      if(nrow(sbtools::item_list_files(ts_id)) > 0 && is.null(item_get(ts_id)$identifiers)) break
-      if(wait==100) stop("identifiers didn't disappear and so can't be replaced; try again later with ",
-                        "repair_ts('", ts_path$var_src, "', '", ts_path$site_name, "')")
+    if(expect_id_loss) {
+      for(wait in 1:100) {
```

Comment:
i agree that functionality like this does belong in sbtools. it's a little beyond what i'd like to implement myself, at least in the next few weeks. i also think maybe for this particular task it's possible that `item_upload_create` would be more appropriate, at least for new ts items. there's also an uploadAndUpdateItem call in the SB API, but i think that still needs a wrapper in SB. and then there are the asynchronous / multi-item options in the SB API, which i don't think are implemented in sbtools at all yet. maybe we could create some issues for all this stuff in sbtools.